### PR TITLE
[Internal] Simplify size computations for some views

### PR DESCRIPTION
**Pull request by pubiqq**

### CODE REVIEW

File: lib/java/com/google/android/material/floatingactionbutton/FloatingActionButton.java [hunk 1]
```diff
@@ -912,32 +913,6 @@ public Drawable getContentBackground() {
...
-      case MeasureSpec.UNSPECIFIED:
-        // Parent says we can be as big as we want. Just don't be larger
-        // than max size imposed on ourselves.
-        result = desiredSize;
-        break;
-      case MeasureSpec.AT_MOST:
-        // Parent says we can be as big as we want, up to specSize.
-        // Don't be larger than specSize, and don't be larger than
-        // the max size imposed on ourselves.
-        result = Math.min(desiredSize, specSize);
```
Comment (drchen) [inline lib/java/com/google/android/material/floatingactionbutton/FloatingActionButton.java:929]: The behavior seems slightly different here? TBH I'm not clear why this method was created at the very beginning. Let me ask some people who may have more knowledge about this.

File: lib/java/com/google/android/material/navigationrail/NavigationRailMenuView.java [hunk 2]
```diff
@@ -63,9 +63,8 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
     }
 
     // Set view to use parent width, but wrap all item heights
-    int parentWidth = MeasureSpec.getSize(widthMeasureSpec);
     setMeasuredDimension(
-        View.resolveSizeAndState(parentWidth, widthMeasureSpec, /* childMeasuredState= */ 0),
+        MeasureSpec.getSize(widthMeasureSpec),
         View.resolveSizeAndState(measuredHeight, heightMeasureSpec, /* childMeasuredState= */ 0));
```
Comment (drchen) [inline lib/java/com/google/android/material/navigationrail/NavigationRailMenuView.java:68]: Can just be `View.resolveSize(measuredHeight, heightMeasureSpec)`?

File: lib/java/com/google/android/material/floatingactionbutton/FloatingActionButton.java (hunk 1 above)
Comment (pubiqq) [reply lib/java/com/google/android/material/floatingactionbutton/FloatingActionButton.java:929]: Yes, it is, but for all acceptable arguments, `View.resolveSize` will return the same values.

File: lib/java/com/google/android/material/navigationrail/NavigationRailMenuView.java (hunk 2 above)
Comment (pubiqq) [reply lib/java/com/google/android/material/navigationrail/NavigationRailMenuView.java:68]: No, because the `MEASURED_STATE_TOO_SMALL` flag will be lost.

Comment (drchen) [reply lib/java/com/google/android/material/navigationrail/NavigationRailMenuView.java:68]: Ah you are right. : )

File: lib/java/com/google/android/material/floatingactionbutton/FloatingActionButton.java (hunk 1 above)
Comment (drchen) [reply lib/java/com/google/android/material/floatingactionbutton/FloatingActionButton.java:929]: Ah you are right. MEASURED_STATE_TOO_SMALL will be removed by View.resolveSize()

**Review by drchen — APPROVED** [review]

**Pull request closed without merging**